### PR TITLE
Kong 3.3.0 UBI 8.7

### PR DESCRIPTION
**Pull request by sumitd2**
## Checklist

- [X] Have you checked and followed all the points mention in the [CONTRIBUTING.MD](https://github.com/ppc64le/build-scripts/blob/master/CONTRIBUTING.md)
- [X] Did you run the script(s) on fresh container with `set -e` option enabled and observe success ?
- [x] Did you have **Legal approvals** for patch files ?

### CODE REVIEW

**Comment by seth-priya** [comment]
@sumitd2 looks like this is failing at the pre-validation stage which probably means that the information in the script header is not consistent, please check, travis should pass for this one I believe

**Comment by sumitd2** [comment]
@seth-priya Its not the header, seems to be that usual problem. And there is another kong PR (Ubuntu):
https://github.com/ppc64le/build-scripts/pull/3225

File: k/kong/kong-3.3.0.patch [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+-	fi
++	$(BAZEL) build //build:venv --verbose_failures --action_env=BUILD_NAME=$(BUILD_NAME)
+ 
+ install-dev-rocks: build-venv
+ 	@. $(VENV) ;\
+@@ -86,7 +84,6 @@ install-dev-rocks: build-venv
+ dev: build-venv install-dev-rocks bin/grpcurl
+ 
+ build-release: check-bazel
+-	$(BAZEL) build clean --expunge
```
Comment (seth-priya) [inline k/kong/kong-3.3.0.patch:29]: why are we removing this?

File: k/kong/kong-3.3.0.patch [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+ package/deb: check-bazel build-release
+diff --git a/build/nfpm/repositories.bzl b/build/nfpm/repositories.bzl
+index cc719072e..8939bc929 100644
+--- a/build/nfpm/repositories.bzl
++++ b/build/nfpm/repositories.bzl
+@@ -13,8 +13,8 @@ def _nfpm_release_select_impl(ctx):
+ 
+     if os_arch == "aarch64":
+         os_arch = "arm64"
+-    elif os_arch == "amd64":
```
Comment (seth-priya) [inline k/kong/kong-3.3.0.patch:41]: why are we overwriting this?

File: k/kong/kong-3.3.0.patch [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
++    elif os_arch == "ppc64le":
++        os_arch = "ppc64le"
+     else:
+         fail("Unsupported arch %s" % os_arch)
+ 
+@@ -36,15 +36,16 @@ nfpm_release_select = repository_rule(
+ 
+ def nfpm_repositories():
+     npfm_matrix = [
+-        ["linux", "x86_64", "4c63031ddbef198e21c8561c438dde4c93c3457ffdc868d7d28fa670e0cc14e5"],
```
Comment (seth-priya) [inline k/kong/kong-3.3.0.patch:52]: why do we need to remove the x86_64 references here and in subsequent lines? can't we just introduce Power support?

File: k/kong/kong-3.3.0.patch [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+         )
+diff --git a/build/nfpm/rules.bzl b/build/nfpm/rules.bzl
+index d6f5bb94f..84ce36ef3 100644
+--- a/build/nfpm/rules.bzl
++++ b/build/nfpm/rules.bzl
+@@ -9,8 +9,8 @@ def _nfpm_pkg_impl(ctx):
+     env = dicts.add(ctx.attr.env, ctx.attr.extra_env, KONG_VAR, ctx.configuration.default_shell_env)
+ 
+     target_cpu = ctx.attr._cc_toolchain[cc_common.CcToolchainInfo].cpu
+-    if target_cpu == "k8" or target_cpu == "x86_64" or target_cpu == "amd64":
```
Comment (seth-priya) [inline k/kong/kong-3.3.0.patch:76]: same thing here, why are we overwriting the arch? seems like this patch will break the x86 builds?

**Review by seth-priya — CHANGES_REQUESTED** [review]

File: k/kong/kong-3.3.0.patch (hunk 1 above)
Comment (sumitd2) [reply k/kong/kong-3.3.0.patch:9]: expunge is no longer supported by bazel, removed from master as well

**Review by seth-priya — APPROVED** [review]

**Pull request merged**